Create Devcontainer config

garrettw opened the pull request:
As requested by @TimWolla.

garrettw committed:
created basic devcontainer config

garrettw committed:
fully automated devcontainer integration

garrett-whitehorn-cengage commented:
@Girgias - in case you'd like to take a look

Girgias commented:
I don't know anything about containerization and don't have any strong opinions about this.

jordikroon committed:
Merge branch 'master' into master

jordikroon committed:
Update Dependabot configuration

jordikroon committed:
run devcontainer without docker

jordikroon committed:
add Java LTS to devcontainer and remove redundant apt install

jordikroon committed:
streamline devcontainer setup and build process

jordikroon committed:
use docker (go back to start pretty much)

jordikroon committed:
upgrade debian

jordikroon committed:
replace Apache with PHP built-in server in devcontainer

jordikroon committed:
Drop LFS

jordikroon committed:
move php server startup to VS Code debug config

jordikroon committed:
improve devcontainer launch configurations and serve script

jordikroon committed:
don't use docker-in-docker

TimWolla commented on .devcontainer/devcontainer.json:
```diff
@@ -0,0 +1,59 @@
+{
+  "name": "PHP",
+  // Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
+  "image": "mcr.microsoft.com/devcontainers/base:bookworm",
```
A `trixie` base should come with PHP 8.4 without needing external repositories.

TimWolla replied:
Ah and of course you pushed right when I commented 😅

jordikroon committed:
enhance devcontainer configuration

jordikroon committed:
improve devcontainer build configuration

jordikroon committed:
increase JVM entity size limits for doc-base builds

jordikroon committed:
increase php memory limit and disable xdebug in devcontainer

jordikroon committed:
correct path for removing xdebug

jordikroon committed:
configure devcontainer memory requirements

jordikroon committed:
chore: remove devcontainer memory requirements

Remove the hostRequirements configuration that specified 4 CPUs and 16GB of memory. This allows the devcontainer to use default resource allocations instead of requiring elevated system resources.

jordikroon committed:
chore: remove symbolic link setup from devcontainer post-create script

The symbolic link creation for doc-base's language source directory is no longer needed and has been removed from the post-create initialization script.

jordikroon committed:
Merge branch 'master' into garrettw/master

jordikroon committed:
Revert "chore: remove symbolic link setup from devcontainer post-create script"

This reverts commit 0b1aa64fedb56d6a9ff251223535f5e6917ee0b4.

TimWolla reviewed (approved):
Did a superficial review, but didn't see any obvious issues. Given the number of commits, I trust that you tested this. Thank you!

jordikroon committed:
fix dependabot.yml indentation
